### ISSUE TITLE
Question about events or push notification

### ISSUE THREAD

**Issue by prasdeva** [grpc/grpc#8718]
Hello,

For simplicity, I have got 1 server and 1 client. From time to time, server need to send notification(or precisely events which it receives from other module) to the client.

Quick skim through the docs tell me that I can use a async client -> server method, where in server will block until the event. Once server receives event, it will unblock and return with the event info. Client would have to post new request to read next event. I verified this functionality with simple test program however have some concerns with performance. 

- There will always be a blocking threads(irrespective of events which are async in nature), one in client and one at server (that to a gRPC thread which might affect gRPC framework). 
- would there any timeouts at client/server in such scenario ?

In other words, I was looking for callbacks or equivalent for my solution. 
I would like to know if this can be done in any other way with in gRPC, more efficiently ? 
Are there any serious downsides of above method which I can't ignore ?

Thanks

**Comment by Hronom** [comment]
Yea, also interested about push notification mechanic with gRPC

**Comment by ctiller** [comment]
Usually best to create a long-lived stream from client -> server to receive
notifications on. The server does writes whenever there's new stuff, and
the client keeps reading.

**Comment by Jacksgong** [comment]
This may be relevant [using grpc for push notification](https://groups.google.com/forum/#!topic/grpc-io/FKeg4yfB-Jo).

**Comment by tigerlee** [comment]
Hope grpc can support push notification ASAP

**Comment by thincal** [comment]
Does this push notification be supported now ?

**Comment by vjpai** [comment]
gRPC has always supported a way to do push notifications. As @ctiller mentioned above, the client should start a long-lived streaming RPC from the server and keep reading on it. Any time the server wants to send a push notification, it should just do a write on that stream.

**Comment by itisbsg** [comment]
https://github.com/itisbsg/grpc-push-notif

this is one implementation of push notifications use case with long lived streams.. probably what @vjpai, @ctiller are referring to.
